components/brave_wallet/browser/meld_integration_service.cc

```diff
@@ -128,6 +132,193 @@ bool NeedsToParseResponse(const int http_error_code) {
   return kRespCodesAllowedToContinueParsing->contains(http_error_code);
 }
 
+void FillCustomerData(
+    const brave_wallet::mojom::CryptoWidgetCustomerDataPtr& customer_data,
+    base::Value::Dict& cbwr) {
+  if (customer_data) {
```

Comment:
Do we want to be modifying this dictionary directly? Seems like an odd pattern in use here where if the data isn't available nothing happens from this function leaving us to error handle elsewhere still.

Reply:
There are 3 calls for Buy, Sell and Transfer widget creation. We retrieve "customer data" from the front-end. It is optional object with set of optional properties, so user can decide to omit it at all.  According to documentation, we can include  "customer data" for each call (buy,sell,transfer). The function `FillCustomerData` has been created to get rid of code duplication. The logic is next: we should not add anything to dictionary in case of "customer data" is `null` and we should not add to dictionary every property, which is `null`.

Reply:
I'm double checking how this data comes in to make sure we're not encountering code paths that could be dangerous from untrusted user supplied data. I think this is fine, but just confirming.

Reply:
Looks like it's not in use yet, so when I'm reviewing front end code for this I'll take a look and might request changes then on frontend or backend. For the time being this doesn't need to block though.